Source/Tests/LinuxSyscalls/Syscalls.h

```diff
@@ -206,6 +218,97 @@ class SyscallHandler : public FEXCore::HLE::SyscallHandler {
   #endif
 
   std::unique_ptr<FEX::HLE::MemAllocator> Alloc32Handler{};
+  
+  ///// VMA (Virtual Memory Area) tracking /////
+
+  FEXCore::HLE::AOTIRCacheEntryLookupResult LookupAOTIRCacheEntry(uint64_t GuestAddr) final override;
+
+
+  struct MRID {
+    uint64_t dev; /* kernel only uses lower 32-bits of dev_t */
+    uint64_t id;
+
+    bool operator<(const MRID& other) const {
+      return std::tie(dev, id) < std::tie(other.dev, other.id);
+    }
+  };
+
+  // We use bits 63:32 for our own custom dev tracking
+  static constexpr const uint64_t MRID_PRIVATE = 0x1'0000'0000ULL;
+  static constexpr const uint64_t MRID_ANON = 0x2'0000'0000ULL;
+  static constexpr const uint64_t MRID_SHM =  0x3'0000'0000ULL;
+
+  struct VMAEntry;
+
+  struct MappedResource {
+    typedef std::map<MRID, MappedResource> ContainerType;
+
+    FEXCore::IR::AOTIRCacheEntry *AOTIRCacheEntry;
+    VMAEntry *FirstVMA;
+    uint64_t Length; // 0 if not fixed size
+    ContainerType::iterator Iterator;
+  };
+
+  struct VMAProt {
+    bool Writable: 1;
+    bool operator==(const VMAProt& other) const {
+      return other.Writable == Writable;
+    }
+  };
+
+  struct VMAOptions {
+    VMAOptions(int Writable, const int Shared) : Prot {Writable != 0}, Flags {Shared != 0} { }
+
+    VMAProt Prot;
+
+    const struct {
+      bool Shared: 1;
+    } Flags;
+  };
+
+  struct VMAEntry {
+    MappedResource* Resource;
+    VMAEntry* ResourcePrevVMA;
```

Comment:
Because the linked list items are also part of an `std::map`, and we'd have store a list of ptrs to the elements and need extra referencing information, which I wanted to avoid.

Though eventually I caved in and added a `MappedResource *Resource` in the `VMAEntry`, so being able to iterate the list directly from a map node is not as important.

Reply:
I think we'll need to convert the linked list to a map eventually, for performance, and just pay the pointer chasing and allocation costs.

Reply:
Sounds reasonable. Maybe worth thinking about adding an explicit `struct VMAEntriesList { }` type to clean up the interfaces a bit (particularly the ListInsert ones, where it's not fully clear which parameter is the new element and which is the list to append to).